### DISCUSSION TITLE
WEBP support ?

### DISCUSSION THREAD

**Discussion by Cipriux** [slatinsky/DiscordChatExporter-frontend#9] [Ideas]
Hi there, very nice app. I'm using it to view the prompts from Midjourney and is working great, but It's getting so much storage space using the images downloaded and I tried converting the png and jpg to webp and also renaming accordingly in the json file, but It doesn't work, the images don't show. It there a reason why it doesn't work ? I do this stuff all the time with my other projects.
Can you please tell me if it should work or not? Or if is possible to add support to also view the webp images also?

**Comment by slatinsky** [comment]
Hello,

webp format is already supported and I verified that embeds in this format work correctly.

The viewer converts files from DiscordChatExporter to its own format. This conversion is expensive, so cache is used. This cache is filename based (so it is fast), but it doesn't detect changes inside the original (json) files. You need to force the rebuild by deleting the cache folder in `dcef\backend\preprocess\temp`. Then close `nginx` window and run `START_VIEWER.bat` again.

**Reply by Cipriux** [reply]
Thank you for the explanations and the fix. The problem was on my part because in installed the viewer on C:\ and inside the "export" folder I put a symbolic link to the another directory on another drive containing the JSON and the images folder. I could only see the messages without the images in the front-end. If I move the JSON and folder directly inside "export" folder, everything is ok, I can see images only. Do you think it should work with symbolic links? If not, I will reinstall the front-end viewer to the same hdd that the messages are. 
Many thanks! 
I wish you to be strong from a Romanian brother and help keeping the world free
[EDIT] Nevermind, I made it work.

**Reply by slatinsky** [reply]
It works with symbolic links too. I use [Link Shell Extension](https://schinagl.priv.at/nt/hardlinkshellext/linkshellextension.html) to create symbolic link of folder inside exports directory.
I do it, because I need to test both dev version and release and exports are really large.

Don't close `nginx` window (local server) while you are using the viewer. Or else additional assets won't load.

**Reply by Cipriux** [reply]
Thanks
